tests/components/homekit_controller/test_light.py

```diff
@@ -152,3 +152,16 @@ def create_lightbulb_service_with_color_temp():
     assert state.state == 'on'
     assert state.attributes['brightness'] == 255
     assert state.attributes['color_temp'] == 400
+
+
+async def test_light_unloaded(hass, utcnow):
+    """Test transition to and from unavailable state."""
+    bulb = create_lightbulb_service_with_color_temp()
+    helper = await setup_test_component(hass, [bulb])
+
+    # Initial state is that the light is off
+    state = await helper.poll_and_get_state()
+    assert state.state == 'off'
+
+    unload_result = await helper.config_entry.async_unload(hass)
+    assert unload_result is True
```

Comment:
We should probably assert that the entity is removed too and any other things are unloaded that should be unloaded.

Reply:
Have fixed the doc string and extended the tests that are performed